src/main/java/com/adobe/cq/commerce/graphql/client/RequestOptions.java

```diff
@@ -78,4 +93,63 @@ public Gson getGson() {
     public HttpMethod getHttpMethod() {
         return httpMethod;
     }
+
+    public CachingStrategy getCachingStrategy() {
+        return cachingStrategy;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+        RequestOptions that = (RequestOptions) o;
+        if (!Objects.equals(httpMethod, that.httpMethod)) {
+            return false;
+        }
+
+        if (CollectionUtils.isEmpty(headers) && CollectionUtils.isEmpty(that.headers)) {
+            return true;
+        }
+        if ((headers == null) ^ (that.headers == null)) { // one is null but not the other
+            return false;
+        }
+        if (headers.size() != that.headers.size()) {
+            return false;
+        }
+
+        // We cannot use Objects.equals with lists because this checks object equality for all list elements
+        // and elements must be in the same order.
+
+        Map<String, String> keyValues = new HashMap<>();
+        headers.stream().forEach(h -> keyValues.put(h.getName(), h.getValue()));
```

Comment:
According to HTTP spec, there can be multiple entries of the same header name (e.g. for cookies) in a single request. Standard Java for example returns headers as `Map<String, List<String>>`. Your comparison here assumes that header names are unique.

Reply:
Good point, I fixed that. Headers are now sorted by name and value, and then compared one by one.